### PR TITLE
new: add fuzzy searching

### PR DESCRIPTION
**Pull request by Wyvest**
## Description
Adds fuzzy searching to OneConfig's GUI search feature.

## Checklist
- [x] I made a clear description of what was changed
- [x] I stated why these changes were necessary
- [x] I updated documentation or said what needs to be updated
- [x] I made sure these changes are backwards compatible
- [x] This pull request is for one feature/bug fix

### CODE REVIEW

File: versions/build.gradle.kts [hunk 1]
```diff
@@ -182,6 +182,10 @@ dependencies {
 
     if (platform.isFabric) {
         include("com.github.Chocohead:Fabric-ASM:v2.3")
+        if (platform.mcVersion <= 11202) {
+            compileOnly(runtimeOnly("org.apache.logging.log4j:log4j-core:2.8.1")!!)
+            compileOnly(runtimeOnly("org.apache.logging.log4j:log4j-api:2.8.1")!!)
+        }
```
Comment (xtrm-en) [inline versions/build.gradle.kts:188]: ?

File: src/main/java/cc/polyfrost/oneconfig/config/elements/OptionSubcategory.java [hunk 2]
```diff
@@ -52,9 +53,9 @@ public OptionSubcategory(String name) {
...
-        List<ConfigPageButton> filteredBottom = bottomButtons.stream().filter(page -> !page.isHidden() && (filter.equals("") || name.toLowerCase().contains(filter) || page.name.toLowerCase().contains(filter) || page.description.toLowerCase().contains(filter))).collect(Collectors.toList());
+        filteredOptions = options.stream().filter(option -> !option.isHidden() && (filter.equals("") || LevenshteinDistance.isSimilar(name, filter) || LevenshteinDistance.isSimilar(option.name, filter))).collect(Collectors.toList());
+        List<ConfigPageButton> filteredTop = topButtons.stream().filter(page -> !page.isHidden() && (filter.equals("") || LevenshteinDistance.isSimilar(name, filter) || LevenshteinDistance.isSimilar(page.name, filter) || LevenshteinDistance.isSimilar(page.description, filter))).collect(Collectors.toList());
+        List<ConfigPageButton> filteredBottom = bottomButtons.stream().filter(page -> !page.isHidden() && (filter.equals("") || LevenshteinDistance.isSimilar(name, filter) || LevenshteinDistance.isSimilar(page.name, filter) || LevenshteinDistance.isSimilar(page.description, filter))).collect(Collectors.toList());
```
Comment (xtrm-en) [inline src/main/java/cc/polyfrost/oneconfig/config/elements/OptionSubcategory.java:58]: these filters look very cluttered

**Review by xtrm-en — DISMISSED** [review]
lgtm

**Review by nxtdaydelivery — APPROVED** [review]
ok

**Pull request merged**